api/clients/accountant.go

```diff
@@ -0,0 +1,158 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	commonpb "github.com/Layr-Labs/eigenda/api/grpc/common"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/meterer"
+)
+
+type IAccountant interface {
+	AccountBlob(ctx context.Context, data []byte, quorums []uint8) (uint32, uint64, error)
+}
+
+type Accountant struct {
+	// on-chain states
+	reservation       core.ActiveReservation
+	onDemand          core.OnDemandPayment
+	reservationWindow uint32
+	pricePerSymbol    uint32
+	minNumSymbols     uint32
+
+	// local accounting
+	// contains 3 bins; circular wrapping of indices
+	binRecords        []BinRecord
+	usageLock         sync.Mutex
+	cumulativePayment *big.Int
+	stopRotation      chan struct{}
+
+	paymentSigner core.PaymentSigner
+}
+
+type BinRecord struct {
+	Index uint32
+	Usage uint64
+}
+
+func NewAccountant(reservation core.ActiveReservation, onDemand core.OnDemandPayment, reservationWindow uint32, pricePerSymbol uint32, minNumSymbols uint32, paymentSigner core.PaymentSigner) *Accountant {
+	//TODO: client storage; currently every instance starts fresh but on-chain or a small store makes more sense
+	// Also client is currently responsible for supplying network params, we need to add RPC in order to be automatic
+	// There's a subsequent PR that handles populating the accountant with on-chain state from the disperser
+	a := Accountant{
+		reservation:       reservation,
+		onDemand:          onDemand,
+		reservationWindow: reservationWindow,
+		pricePerSymbol:    pricePerSymbol,
+		minNumSymbols:     minNumSymbols,
+		binRecords:        []BinRecord{{Index: 0, Usage: 0}, {Index: 1, Usage: 0}, {Index: 2, Usage: 0}},
+		cumulativePayment: big.NewInt(0),
+		stopRotation:      make(chan struct{}),
+		paymentSigner:     paymentSigner,
+	}
+	// TODO: add a routine to refresh the on-chain state occasionally?
+	return &a
+}
+
+// accountant calculates and records payment information
```

Comment:
nit: `BlobPaymentInfo calculates...`

Reply:
can we also document what's being returned?

Reply:
updated the comments